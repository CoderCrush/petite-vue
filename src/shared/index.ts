export * from './toDisplayString'

export const extend = Object.assign

export const EMPTY_OBJ = {}

export const isObject = (value) => {
  return value !== null && typeof value === 'object'
}

export const isString = (value) => {
  return typeof value === 'string'
}

export const hasChanged = (value, newValue) => {
  return !Object.is(value, newValue)
}

export const hasOwn = (value, key) => Object.prototype.hasOwnProperty.call(value, key)

export const camelize = (str) => {
  return str.replace(/-(\w)/g, (_, c) => {
    return c ? c.toUpperCase() : ''
  })
}

const capitalize = str => str.charAt(0).toUpperCase() + str.slice(1)

export const toHandlerKey = str => str ? `on${capitalize(str)}` : ''
