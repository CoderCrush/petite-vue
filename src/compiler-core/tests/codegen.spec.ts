import { generate } from '../src/codegen'
import { baseParse } from '../src/parse'
import { transform } from '../src/transform'

describe('compiler-core/codegen', () => {
  it('string', () => {
    const ast = baseParse('hi')
    transform(ast)
    const { code } = generate(ast)
    expect(code).toMatchSnapshot()
  })
})
