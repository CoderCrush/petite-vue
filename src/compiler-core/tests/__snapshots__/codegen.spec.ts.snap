// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiler-core/codegen element 1`] = `
"
return function render(_ctx, _cache) {return }"
`;

exports[`compiler-core/codegen interpolation 1`] = `
"const { toDisplayString: _toDisplayString } = Vue
return function render(_ctx, _cache) {return _toDisplayString(_ctx.message)}"
`;

exports[`compiler-core/codegen string 1`] = `
"
return function render(_ctx, _cache) {return \\"hi\\"}"
`;
